addons/mail/models/mail_thread.py

```diff
@@ -1043,7 +1043,7 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
                 rcpt_tos_valid_localparts = [to for to in rcpt_tos_valid_localparts if to in other_model_aliases.mapped('alias_name')]
 
```

Comment:
Oh and since it can be done in python, you should also make sure the issue doesn't exist in prior versions. We support 3 versions at all time, so you should check it doesn't exist on 14.0 and up. 

If it does, you can move the fix to that version by either creating a new PR and mentioning this one in it, or (better) resetting your branch to whatever version the bug appears in, and retargeting the PR (just click edit and modify the version at the top). And once it's all reviewed and validated it will automatically port it to all other versions.

Reply:
Yes

You could try replicating on the 14.0 runbot instead of creating one straight away but that's the idea.